agent/src/lib/helpers.ts

```diff
@@ -9,10 +17,121 @@ export function wildcardMatch(target: string, pattern: string): boolean {
     return exp.test(target);
 }
 
+/**
+ * Enhanced debug printing. For XPC dicitionaries we print entire byte arrays instead of just the beginning. This enhances the output compared to the standard output by debugPrinting 
+ * @param ptr Objective C Pointer 
+ * @returns A debug string with all raw data contents 
+ */
 export function objcObjectDebugDesc(ptr: NativePointer) {
-    return (new ObjC.Object(ptr)).toString();
+    // We enhance the debug description by actually printing complete data blobs as base64 instead of trimming the data when using the current implementation 
+
+    let objcObject = new ObjC.Object(ptr);
+    let className = objcObject.$className;  
+    let outString = ""
+
+
+    if (className === "OS_xpc_dictionary") {
+        let entries = xpcDictionaryGetCount.call(objcObject); 
+
+        if (entries > 0) {
+            outString += debugDescriptionForXPCDictionary(objcObject, entries); 
+        }else {
+            outString += objcObject.toString(); 
+        }
+    }else {
+        outString += objcObject.toString(); 
+    }
+
+    return outString;
+}
+
+export function debugDescriptionForXPCDictionary(xpcDict: ObjC.Object, count) {
```

Comment:
It would probably be better to abstract this and `parseBPListKeysRecursively()` into one function since they share most of the code. The whole codebase needs some good factoring anyways, so let me do that in the future sometime.

Reply:
I did not look into too much detail on how the bpList parsing part works. But, I discovered an so far undocumented bplist format. bplist17 seems to be used by newer binaries